### PR TITLE
Support defaulting kubeconfig for CAPI clusters

### PR DESCRIPTION
**Pull request by rossf7**
Fixes https://github.com/giantswarm/giantswarm/issues/16742

Fixes problem found by @calvix with using `kubectl gs template app` with CAPI clusters.

See https://gigantic.slack.com/archives/CQZA6GGAY/p1631267306035400

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by rossf7** [comment]
Tested on ghost.

File: pkg/app/mutate_app.go [hunk 1]
```diff
@@ -255,6 +259,33 @@ func (m *Mutator) mutateLabels(ctx context.Context, app v1alpha1.App, appVersion
 	return result, nil
 }
 
+func findKubeConfigNamespace(ctx context.Context, k8sClient kubernetes.Interface, appNamespace, kubeConfigName string) (string, error) {
+	_, err := k8sClient.CoreV1().Secrets(appNamespace).Get(ctx, kubeConfigName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
```
Comment (piontec) [inline pkg/app/mutate_app.go:264]: Nitpick: is not better to reverse the condition? If it was found return immediately. If not, then search for it. That way you will avoid the nested `if` and the sudden `return` of a value that was set at the beginning of the func.

**Review by piontec — APPROVED** [review]
LGTM, with a nitpick

**Review by calvix — APPROVED** [review]
lgtm

File: pkg/app/mutate_app.go (hunk 1 above)
Comment (rossf7) [reply pkg/app/mutate_app.go:268]: Yes that is better 👍 thanks

**Pull request merged**